submission_worker.py

```diff
@@ -0,0 +1,309 @@
+from __future__ import absolute_import
+import contextlib
+import django
+import importlib
+import json
+import os
+import pika
+import requests
+import shutil
+import signal
+import socket
+import sys
+import time
+import yaml
+import zipfile
+
+from os.path import join
```

Comment:
`join` is used many a time, importing here to avoid writing `os.path.join` again and again.